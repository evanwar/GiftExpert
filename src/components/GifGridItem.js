import React from 'react'
import PropTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {

    return (
        <div className="card  animate__animated animate__fadeInLeft">
            <img alt={title} src={url} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifGridItem;
