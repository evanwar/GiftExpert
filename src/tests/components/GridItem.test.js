import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('Unit Test GridItem', () => {

    const title = 'Overwatch';
    const url = 'https://www.dzoom.org.es/wp-content/uploads/2010/09/mirada-ojos-encuadre-primer-plano-sexy-810x540.jpg';

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('should equal argument', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('should have p whit title', () => {

        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);

    });


    test('should have image equal to Url', () => {

        const image = wrapper.find('img');

        expect(image.prop('src')).toBe(url);
        expect(image.prop('alt')).toBe(title);

    });


    test('debe tener una animacion', () => {

        const div = wrapper.find('div');

        const classStyle = div.prop('className');

        expect(classStyle.includes('animate__animated')).toBe(true);


    });

});