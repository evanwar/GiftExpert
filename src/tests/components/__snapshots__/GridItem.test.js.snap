// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test GridItem should equal argument 1`] = `
<div
  className="card  animate__animated animate__fadeInLeft"
>
  <img
    alt="Overwatch"
    src="https://www.dzoom.org.es/wp-content/uploads/2010/09/mirada-ojos-encuadre-primer-plano-sexy-810x540.jpg"
  />
  <p>
    Overwatch
  </p>
</div>
`;
